File: client/src/app/shared/models/learning_resource_model.ts
import { LearningResource } from './dtos/learning_resource_dto';
import { Post } from './dtos/post_dto';
import { UserResourceProgression } from './dtos/user_resource_progression_dto';

export interface LearningResourceModel extends LearningResource {
  posts: Post[];
  userResourceProgression: UserResourceProgression;
}

File: server/BitBoard/+Business/BitBoard.Business.Learning/Views/LearningResourceModel.cs
using System.Collections.Generic;
using API.Models.DTOs;

namespace BitBoard.Business.Learning.Views
{
    public class LearningResourceModel : LearningResourceDto
    {
        public ICollection<PostDto> Posts { get; set; }
        public UserResourceStateDto UserResourceProgression { get; set; }
    }
}